framework/helpers/BaseStringHelper.php

```diff
@@ -43,7 +45,11 @@ public static function byteLength($string)
      */
     public static function byteSubstr($string, $start, $length = null)
     {
-        return mb_substr($string, $start, $length === null ? mb_strlen($string, '8bit') : $length, '8bit');
+        if ($length === null) {
+            $length = static::byteLength($string);
```

Comment:
We have intentionally avoided method calls there for performance reasons.

Reply:
@samdark i don't see difference
~~~php
public static function byteLength($string)
{
    return mb_strlen($string, '8bit');
}
~~~

Reply:
Technically that's the same bug generates different set of opcodes because of the method call itself. Since `byteLength` could be called thousands of times, it makes sense to make it as performant as possible.

Reply:
@samdark method use another class method - standard practice for any class. When it's change performance?

Reply:
The difference is another method call. It's relatively cheap but isn't free. May not matter at all with modern PHP versions and hardware but did matter when the code was initially written.

Reply:
test of 3000 random strings: with `byteLength()` - 0,008s, without - 0,004s, test of 1000 strings - 0,001 / 0,002s.
I add counter to `byteLength()` - my app do less than two hundred call to method.

Reply:
Yes. For regular apps it doesn't matter much but it my matter for batch export or batch import. It's not uncommon to deal with 10K elements in a single batch doing 2-3 `byteLength()` calls. That may result in 10-20 seconds difference.

Reply:
@samdark class uses `array_*()` with closures - I don't think that's faster then direct calling in cycles. That's code steals more time than `byteLength()` .
~~~php
$titelized = array_map(function ($word) use ($encoding) {
    return static::mb_ucfirst($word, $encoding);
}, $words);
~~~
~~~php
$result = array_values(array_filter($result, function ($value) {
    return $value !== '';
}));
~~~

Reply:
Yeah. That's not optimal performance-wise. But... let it be, not a huge difference for majority of cases anyway.

Reply:
@samdark

> not a huge difference for majority of cases anyway.

Again double standards in action! "10K elements" is minor case? If 3k elements is 0,008sec, then 10k is ~0,03sec, how you calculate "10-20 seconds"?